docs/modeling/changes_for_4.rst

```diff
@@ -1,4 +1,181 @@
+  that they could not hold values for the models. Instead, the 
+  values were held inside the models. This resulted in confusion
+  when compound models were used since this necessitated that
+  the compound models make copies of the values. As a result, 
+  changing the value in the compound model did not change the
+  constituent model's parameter value and visa versa. Now
+  parameters are distinct instances for each use, and they
+  do now hold the value of the parameter. So compound models
+  now share the same values as the constituent models.
+
+- Previously when model sets were used, the parameter shape
+  did not show the corresponding dimension for the number 
+  of models. Now it does. For example:
+
+  Old::
+
+        In [1]: g = Gaussian1D([1,1], [1,2], [2,4], n_models=2)
+        In [2]: g.amplitude
+        Out[2]: Parameter('amplitude', value=[1. 1.])
+        In [3]: g.amplitude.shape
+        Out[3]: ()
+
+  New::
+
+        In [1]: g = Gaussian1D([1,1], [1,2], [2,4], n_models=2)
+        In [2]: g.amplitude
+        Out[2]: Parameter('amplitude', value=[1. 1.])
+        In [3]: g.amplitude.shape
+        Out[3]: (2,)
+
+- Previously the values were held in an array within the model
+  instance and it was possible to assign values to slices of
+  that array. Reassigning the array does update the parameters,
+  but assigning slices does not. The new approach is to either
+  replace the whole array by assigning to the parameters property
+  or assign directly to the parameter value.
+
+- The use of 'imputed' units, i.e., supplying input/output units
+  to a compound model without them but where the component models 
+  support the _parameter_units_for_data_units() method is much more
+  restricted in its applicability. That will only work when the
+  compound expression uses the '+' or '-' operators. Past behavior
+  led to sometimes arbitrary assignments of units, and sometimes
+  incorrect units to the parameters.
+
+- Slicing is more restrictive now. Previously a model defined as 
+  such::
+
+        m = m1 * m2 + m3
+
+  permitted this slice::
+
+        m[1:] # results in m2 + m3
+
+  Now, only submodels in the expression tree (think of it as
+  the sequence of operations as performed) are permitted as 
+  slices. This means some slices that make sense do not work
+  now. The following code illustrates what is permitted 
+  and what isn't::
+
+        m = m1 + m2 + m3 + m4
+        m[:2] # Results in m1 + m2, works
+        m[1:] # Should result in m2 + m3 + m4, does not work
+               # since m1 is part of all subexpressions.
+
+
+
+- Generally, all public methods have remained unchanged. The exception
+  is .n_submodels that used to be a method but now is a property.
+
+- Many of the non-public methods have changed, particularly for
+  compound models.
+
+- The _CompoundModelMeta metaclass no longer exists.
+
+-----------------
+Technical Details
+-----------------
```

Comment:
This is not the standard way to do sub-headings, I think except for the top-level one it should just be one line below (not above). Can you check another page to make sure?

Reply:
The standard for headings is just the line below. I'm not sure if the line above actually changes the hierarchy rendering but I suggest taking them out for consistency!